serpentTools/parsers/bumat.py

```diff
@@ -12,4 +12,101 @@ class BumatReader(MaterialReader):
     filePath: str
         path to the depletion file
     """
-    pass
+
+    def __init__(self, 
+
+    if materialname==None or materialfile==None:
+        raise Exception("A material file and material name in that file must be specified in order to read in serpent material data as a salt.")
+    debugserpentoutput=False #debug
+    self.materialname=materialname
+
+    with open(self.directory+'/'+materialfile, 'r') as bumat_file:
+
+        #this for loop will iterate through all lines in the
+        #burned material output
+        on_correct_material=False #used to see if the line has isotopes fractions to be read in
+        for line in bumat_file.readlines():
+            if '%' ==line[0]:
+                continue #skip comments
+            #see if the line defines a material. if it is the one
+            #of interest, record its isotopics, atom density, temp
+            #library, etc.
+
+            #firstly, if the correct material was already found,
+            #and the current line is a new material definition,
+            #then this loop is done reading for the material of
+            #interest.
+
+            if (line[:3]=='mat' or line[:3]=='set') and on_correct_material:
+                if debugserpentoutput:
+                    print("finished finding material {0}".format(materialname))
+                break #leave the for loop, material was found
+            if line[:3]=='mat':
+
+                line.rstrip() #remove trailing whitespace
+                #split the line to read in separate words.
+                splitline=line.split() #now is list data type
+                thismaterialname=splitline[1] #grab material name
+                if debugserpentoutput:
```

Comment:
Should be a call to [`messages.debug`](https://github.com/CORE-GATECH-GROUP/serpent-tools/blob/develop/serpentTools/settings/messages.py#L50)

Reply:
thanks! very sophisticated!